### ISSUE TITLE
Meteor crashes constantly with EPERM on Windows

### ISSUE THREAD

**Issue by stubailo** [meteor/meteor#4933]
Originally posted by @fjafjan on the windows-preview repo

I am building a fairly rather simple Meteor project on Windows 8, and despite both re-installing meteor, and manually re-setting the library, after the app has run for a short time meteor crashes and will not restart. Meteor is also unable to run reset, and I haven't been able to identify exactly what it is that causes this arise in the first place. There is a consistent error I get from the ccorcos:swipe, but runtime errors should not be crashing meteor right?

I have tried changing/removing the files in .meteor/client.

My question is both to identify what could be causing this particular error (what does "stat" mean in this context? other error messages I have found have while searching have been far more useful) as well as generally if there is a faster way of hard resetting a meteor project than running meteor create proj_name in a new folder and manually moving over the source files. 

The stacktrace is below.

The packages I use are 

 - meteor-platform 
 - insecure 
 - bevanhunt:leaflet 
 - mdg:geolocation
 - accounts-facebook
 - service-configuration 
 - less stevezhu:lodash
 - ccorcos:swipe
 - iron:router 
 - underscore 
 - coffeescript 
 - accounts-ui
 - accounts-password

```
    C:\Users\Fjafjan\Documents\GitHub\progress>meteor run
    [[[[[ C:\Users\Fjafjan\Documents\GitHub\progress ]]]]]

    => Started proxy.
    => Started MongoDB.

    C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\dev_bundle\lib\node_modules\fibers\future.js:245
                                                    throw(ex);
                                                          ^
    Error: EPERM, stat 'C:\Users\Fjafjan\Documents\GitHub\progress\.meteor\local\build\programs\web.browser\packages'
      at Object.Future.wait (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\dev_bundle\lib\node_modules\
    fibers\future.js:398:15)
      at Object.wrapper [as stat] (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\files.js:1350:24
    )
      at C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\files.js:454:43
      at Array.forEach (native)
      at Function._.each._.forEach (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\dev_bundle\lib\node_m
    odules\underscore\underscore.js:79:11)
      at Object.files.cp_r (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\files.js:444:5)
      at C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\files.js:456:13
      at Array.forEach (native)
      at Function._.each._.forEach (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\dev_bundle\lib\node_m
    odules\underscore\underscore.js:79:11)
      at Object.files.cp_r (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\files.js:444:5)
      at C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\files.js:456:13
      at Array.forEach (native)
      at Function._.each._.forEach (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\dev_bundle\lib\node_m
    odules\underscore\underscore.js:79:11)
      at Object.files.cp_r (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\files.js:444:5)
      at Object.files.rename (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\files.js:1401:13)
      at Object.files.renameDirAlmostAtomically (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\fi
    les.js:810:11)
      at [object Object]._.extend.complete (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\builder
    .js:482:11)
      at C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\bundler.js:1933:13
      at C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\bundler.js:2147:20
      at C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\buildmessage.js:264:13
      at [object Object]._.extend.withValue (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\fiber-
    helpers.js:115:14)
      at C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\buildmessage.js:257:29
      at [object Object]._.extend.withValue (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\fiber-
    helpers.js:115:14)
      at C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\buildmessage.js:255:18
      at [object Object]._.extend.withValue (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\fiber-
    helpers.js:115:14)
      at C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\buildmessage.js:246:23
      at [object Object]._.extend.withValue (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\fiber-
    helpers.js:115:14)
      at Object.capture (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\buildmessage.js:245:19)
      at Object.exports.bundle (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\bundler.js:2040:31)

      at C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\run-app.js:554:24
      at time (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\profile.js:231:28)
      at Function.run (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\profile.js:377:12)
      at bundleApp (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\run-app.js:544:34)
      at [object Object]._.extend._runOnce (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\run-app
    .js:591:35)
      at [object Object]._.extend._fiber (C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\run-app.j
    s:821:28)
      at C:\Users\Fjafjan\AppData\Local\.meteor\packages\meteor-tool\1.1.3\mt-os.windows.x86_32\tools\run-app.js:402:12
```

**Comment by stubailo** [comment]
Originally posted by @stumacd:

I am getting the same issue on Windows 7 - 64 bit machine.

**Comment by stubailo** [comment]
Originally posted by @sdarnell:

In essence the code is trying to build a bundle ready for execution. During this process it is trying to be a good citizen by ensuring that you don't end up with a partial bundle in the case of an error.
It does this by building in a temporary folder, then 'atomically' renaming the folder to the final destination. (That's the renameDirAlmostAtomically in the stack trace)

This approach works well on unix like platforms, but doesn't always work well on Windows. The symptom is that when meteor tries to rename the folder, it is still in use, and you get an EPERM (permission denied error). So on Windows meteor has a fallback, and resorts to copying then deleting the original.
(That's the cp_r in the stack)

The copy should succeed, but for some reason it hasn't in this case - not sure why. Your line numbers don't quite line up with the current sources, so I'm not sure. But before it copies a file, it first check that it can 'stat' it. That is, read its file metadata.

Some scenarios can produce an EPERM if the file is 'scheduled for deletion', and this may be what is happening here - but it is difficult to tell.

As for resetting quicker, there is the `meteor reset` command, which should be really quick.

**Comment by phocks** [comment]
I can confirm that a similar problem is happening in Windows 10. It doesn't happen all the time, but seemingly at random when saving and Meteor is refreshing. Here is my error `Error: EPERM, unlink`

```
C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\dev_bundle\lib\node_modules\fibers\future.js:278
                                                throw(ex);
                                                      ^
Error: EPERM, unlink 'C:\Users\Joshua\Dropbox\Code\qurious\.meteor\local\build\programs\web.browser\packages\blaze.js'
    at Object.Future.wait (C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\dev_bundle\lib\node_modules\fibers\future.js:398:15)
    at Object.rm_recursive (C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\files.js:268:9)
    at Object.files.rename (C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\files.js:1402:13)
    at Object.files.renameDirAlmostAtomically (C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\files.js:810:11)
    at [object Object]._.extend.complete (C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\builder.js:482:11)
    at C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\bundler.js:1819:11
    at C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\bundler.js:2142:11
    at Function._.each._.forEach (C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\dev_bundle\lib\node_modules\underscore\underscore.js:87:22)
    at C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\bundler.js:2140:9
    at C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\buildmessage.js:264:13
    at [object Object]._.extend.withValue (C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\fiber-helpers.js:115:14)
    at C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\buildmessage.js:257:29
    at [object Object]._.extend.withValue (C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\fiber-helpers.js:115:14)
    at C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\buildmessage.js:255:18
    at [object Object]._.extend.withValue (C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\fiber-helpers.js:115:14)
    at C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\buildmessage.js:246:23
    at [object Object]._.extend.withValue (C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\fiber-helpers.js:115:14)
    at Object.capture (C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\buildmessage.js:245:19)
    at Object.exports.bundle (C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\bundler.js:2040:31)
    at C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\run-app.js:554:24
    at time (C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\profile.js:231:28)
    at Function.run (C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\profile.js:377:12)
    at bundleApp (C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\run-app.js:544:34)
    at [object Object]._.extend._runOnce (C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\run-app.js:750:35)
    at [object Object]._.extend._fiber (C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\run-app.js:821:28)
    at C:\Users\Joshua\AppData\Local\.meteor\packages\meteor-tool\1.1.4\mt-os.windows.x86_32\tools\run-app.js:402:12
    - - - - -
```

**Comment by sjmcdowall** [comment]
That Dropbox looks suspicious -- does this happen if you have your project on a local disk under your profile area (C:\Users\Josua\Code for example) ?

**Comment by quetzalcoatl** [comment]
Hello! I also noticed this issue on my very small "project" I created for learning meteor. I have pretty clean setup,  it wasn't crashing from the start like that. I actually don't remember when it has started crashing, and similar to others, I thought at first that this something with code editor (gVim) that likes to create lock files, but no, this even happens if I edit files with plain ol' notepad.

Today, I noticed that this issue occurrs **only** when I have some unit/integration-tests with velocity. When I move the `tests` folder out of the project, or if I change its name to `test--` or whatever similar, Velocity stops processing this files (of course), sees no tests, mirror nodejs server doesn't start, and the issue does not occurr at all.

I've got:

- meteor-platform              1.2.2
- sanjo:jasmine                0.17.0
- velocity:html-reporter       0.8.2
 - ^ those are probably relevant, and some other:
- accounts-password            1.1.1
- accounts-ui                  1.1.5
- urigo:angular                0.9.3
- angularui:angular-ui-router  0.2.15

**Comment by quetzalcoatl** [comment]
After some digging I've read [this post on SO] (http://stackoverflow.com/questions/14508933/enotempty-and-eperm-exceptions-while-bundling-meteor-application-on-win7) which suggests that similar issues were already seen, diagnosed and fixed. I've checked my meteor installation for RIMRAF and I've noticed that:

- in meteor-tool 1.1.4 in LIB\node_modules there's RIMRAF 2.2.8 and GLOB 3.2.1
- in meteor-tool 1.1.4 in BIN\node_modules there's RIMRAF 2.3.2 and GLOB 4.4.2

I've added some debug logs and the EPERM was raised from the version from \LIB.

So, I've tried "upgrading" (I mean, manually copying-replacing) the newer versions from BIN into LIB, also I had to tag INFLIGHT package along. Problem still occurs, but seems to happen much less frequently.

----

In the newer, RIMRAF 2.3.2, when a directory is removed and the `unlink` from `rimraf_` throws EPERM, windows is detected and `fixWinEPERM` is called. There, there's a `chmod` + `stat` pair that attempts to fix things. However, if `chmod` fails not due to ENOENT, then the callback passes the original error (from unlink) to the caller. The "caller" is the wait&retry loop, in `rimraf.js` in `afterGlob`, and it contains a check:

          if (isWindows && (er.code === "EBUSY" || er.code === "ENOTEMPTY") &&
              busyTries < options.maxBusyTries) {
            busyTries ++
            var time = busyTries * 100
            // try again, with the same exact callback as this one.
            return setTimeout(function () {
              rimraf_(p, options, CB)
            }, time)
          }

once that EPERM gets here, this wait&retry does not attempt to retry it, returns it as unhandled, hence crash. Adding EPERM to that retry condition seems to solve `EPERM, unlink` case. By the way, his was the one I observed almost all of the times. Btw2: Of course still can crash if 3 retries doesn't suffice, but that's how RIMRAF works currently.

**Comment by quetzalcoatl** [comment]
---

The second thing, the `EPERM, stat / web.browser`  is not raised in RIMRAF, but from fileops invoked during `renameDirAlmostAtomically`. It is based on `files.rename` which contains some basic safeguards against Windows and "BUSY" error, performs some retrying, and finally falls back to `cp_r` and `rm_recursive` (RIMRAF).

The problem here is that if 'rename' failed due to, say, intermittent EPERM :), then `stat` or other ops in `cp_r` can fail the same too. Probably the most important thing here is that that the loop that guads against "BUSY" (actually, it catches just any failure) performs retries **immediately**. RIMRAF has a saner approach - repeats each attempt with expanding delay value. Retrying immediatelly probably doesn't give enough time to EPERM on a folder to disappear.

**Comment by quetzalcoatl** [comment]
I think I've solved that `EPERM, stat`. I've improved the special case for windows in `files.rename(from,to)` in `files.js` with EPERM handling at cp_r/rm_recursive and also added expanding wait&retry similar to what is found in RIMRAF.

After those two fixes (eperm at rimraf earlier, and now this fixes to rename), issue seems to be gone. However I find it a little strange that this whole fallback to cp-r/rm-r is needed in rename during code rebuild after detecting a change. I mean, who's holding onto that files and preventing windows from moving the directory?

**Comment by fjafjan** [comment]
I posted about this issue on stackoverflow and someone responded that it
was solved if they closed their editor when compiling, so that is probably
what is holding onto the file? Can't say I know much about windows file
management though

**Comment by fjafjan** [comment]
Or rather, when installing a package.
http://stackoverflow.com/questions/30380268/meteor-crashing-while-building?noredirect=1#comment51995396_30380268

**Comment by sdarnell** [comment]
I think the reason the previous code retried immediately is because it wasn't always in a fiber so couldn't pause. But the interface is blocking so you couldn't register a timeout and return etc. I couldn't find any way to sleep in node. Even the sync form of rimraf doesn't seem to sleep+retry. 
There is the node.sleep npm module which is the only option - other than fibers of course.

**Comment by quetzalcoatl** [comment]
Actually, I had no big problems in introducing a wait with fibers there, and it seems to be working. I've left the old way as a fallback when fibers are not available.

However, I've noticed that adding that better-waiting does not help much. Files/folders are still locked and not removable and regardless of how long delay and how many attempts I set, the process always falls back to to cp-r/rm-r fallback.

What really helps here is just the handling of EPERM condition so the proces retries instead of crashing instantly.

I'm not very fond of those two patches. Fixing things by adding more delays usually isn't a good idea :) If I knew node and meteor better, I'd try to find another way. I think it'd be the best to ensure that these locks are released before trying to rename/remove foders, but I don't know if it is possible, I don't know where the locks come from in the first place.

**Comment by sdarnell** [comment]
Sorry I probably wasn't clear. I was just trying to justify the busy retry aspect. Note that it was over a year ago, and I seem to recall the main uses were sync without fiber. A lot has changed since then, and I suspect the majority of callers are now inside a fiber.

The problem is that Windows will not let you remove files/directories that are in use - it just won't, and this is very different to Unix-like platforms in that regard. Modern editors, windows search, virus scanners etc. all like to visit new directories and files, and they seems to be the main culprit for holding files open. Though it is also common for people to end up having the generated files loaded in an editor.

These issues are exacerbated by the Meteor packaging system using multiple steps to build the final files, and using the 'rename almost atomically' to move them about. Essentially the more times this is used on freshly created (and ripe for scanning) files, the more likely the symptoms will occur.

**Comment by quetzalcoatl** [comment]
Ok, I understand what you mean. I didn't have any editor at that time, and I have no active antivirus. Maybe the windows search host service.. However, I'm pretty sure I caught it once with ProcessHacker file handle search and saw that filelocks coming from the nodejs process. Also the fact that when I extended the timeouts/retries up to the range of minutes, the EPERM condition never disappeared and the rename-dir-almost-a still felt back to copy-remove. I'm pretty sure that windows-search scanner and whatever I have on my OS would release the locks by that time. This is why I think that there is some internal problem in Meteor (re/auto)build process.

btw. do you think that #4980 is worth polishing any further? Aside from the master/devel catch I need to change, it's still only a partial fix, it won't be working in its best form without https://github.com/isaacs/rimraf/issues/86 and updating the deps to the newest rimraf, assuming they agree to that change. I'm not sure how to coordinate that. I could try lifting the PERM@unlink fix from rimraf's to meteor's codebase, what do you think?

**Comment by sdarnell** [comment]
RE: Potential internal problem in Meteor
I would expect this sort of thing to be a bit more reproducible that it seems to be, however, it is clearly a possibility.
One easy to spot problem is if the node process' current directly gets left in one of the build directories. When the issue occurs, just check the current dir.
Another is whether a file handle gets left open to a file in one of the build directories, this is a bit trickier.
The thing here, a good thing to try to establish is whether this form of failure is persistent or goes away. If it is persistent, there's a bit more scope to attach process debugging tools to inspect handles, or even debug into the node data structures. Maybe adding some code to wrap the file opening/closing methods and to track orphaned objects at the time of a failure. (fs/files.js already wraps most entry points in some way).

As for the change, I had a quick look and seems sensible. There is already a version of sleepMs in utils which should be used rather than adding another version. It really needs the core team to check though, as the code base has changed a lot since I was familiar with it.
Sounds like updating the rimraf version used in all cases makes sense.

**Comment by quetzalcoatl** [comment]
I have just found out the source of EPERMs in my case - sanjo:jamsine uses Karma to run tests, Karma in turn uses Chokidar. Chokidar seems to keep a permanent handle to each observed directory.  This literally completely prevents renaming directories and that's why I had observed that all renames fall back to cp-r/rm-r. I still don't fully understant why the rm-r (rimraf) in fallback succeeds at some point. Also, it seems that it does not observe directories too well - https://github.com/paulmillr/chokidar/issues/324 Oh, and I'm still talking about Windows - I have no idea how's the situation on Linux, probably much better as the FS behaves differently.

**Comment by phocks** [comment]
I believe this issue is definitely related to Windows or a program within Windows "watching" the folder and thus preventing renaming, something like a virus scanner or folder sync. When I moved my working folder out of Dropbox and to an unwatched folder the issue disappeared almost completely. It has only happened once or twice afterwards. The issue may be dependant on how large your application is also, as I never used to get the error even when my working folder was in Dropbox when I first started working on it and it was small.

It appears that Windows places only brief temporary locks on these files. Maybe if Meteor could wait a little while and then try again in Windows instead of throwing an error and completely exiting, then this issue might be resolved.

**Comment by sebakerckhof** [comment]
Also experiencing this in 1.2-rc.12, partial trace:
```
C:\Users\seke\AppData\Local\.meteor\packages-from-server\kndedu01.barco.comX3005\meteor-tool\1.1.7-rc.12\mt-os.windows.x86_32\dev_bundle\lib\node_modules\fibers\future.js:245                        
                                                throw(ex);                                                                                                                                            
                                                      ^                                                                                                                                               
Error: EPERM, unlink 'D:\tmp\newmeteor\test\.meteor\local\isopacks\fourseven_scss-garbage-1e1k1sw\plugin.compileScssBatch.os.windows.x86_32\npm\compileScssBatch\node_modules\node-sass\vendor\win32-i
2-11\binding.node'                                                                                                                                                                                    
    at Object.Future.wait (C:\Users\seke\AppData\Local\.meteor\packages-from-server\kndedu01.barco.comX3005\meteor-tool\1.1.7-rc.12\mt-os.windows.x86_32\dev_bundle\lib\node_modules\fibers\future.js:
8:15)                                                                                                                                                                                                 
    at Object.rm_recursive (C:\tools\fs\files.js:270:9)                                                                                                                                               
    at Object.files.renameDirAlmostAtomically (C:\tools\fs\files.js:824:11)                                                                                                                           
    at Builder.complete (C:\tools\isobuild\builder.js:571:13)                                                                                                                                          
....                                                                                                                                                                    
```

**Comment by stubailo** [comment]
So can we just fix this by upgrading rimraf now that your PR is accepted? Or is there something else @quetzalcoatl

**Comment by quetzalcoatl** [comment]
Yes, I think so. I'm using this patch for quite a while now and I didn't notice this issue anymore,

**Comment by sebakerckhof** [comment]
Please re-open or fix this: https://github.com/meteor/meteor/issues/5278
Because the new behavior can be even more annoying than the old one, since you don't even know if the build is just taking long or if it will take 'forever'

**Comment by benjamn** [comment]
@sebakerckhof let's keep this closed and focus on #5278 since you've identified a slightly different root cause behind that issue
